Update reference generation instructions for p5.js 2.0 fork workflow

adi-ray opened the pull request:
### Description:
PR updates the outdated contributor documentation which was still referencing the old workflow using `npm run docs` and `npm run docs:dev` commands within the p5.js repository.

> [!NOTE]
>  Please review the English content changes first. Once approved, I'll be happy to update the other language versions accordingly.

Fixes #924

yinhwa commented:
@montoyamoraga much'd be appreciated if you could spare time for reviewing this aaronee 😁😁 Thank you!

montoyamoraga reviewed (commented):
hi @yinhwa!

thanks for making this more manageable pull request!
these are my comments on the 7 files:

OK: 
* i18n-tracking.yml: i understand this is automated, so i think it's OK
 * src/templates/pages/teach/index.hbs: this is where i think you did most of your work, and the code looks good to me, but i don't know how to build it without merging the PR.

Doubts:
main.css, en.yml / es.yml / ko.yml / zh-Hans.yml 
these files confused me, because it seems they have many changes, some of them i'm guessing are unnecessary and might have been done by running grunt, or by your code editor, so i think we need help with these.

i am going to ask for review to @limzykenneth so we can have help :)

limzykenneth commented:
I assume the big diff is more due to the `teach`  branch being a bit outdated and @yinhwa you have rebased before working on this latest set of changes? If that's the case it should be fine. We can do one big final review when the `teach` branch is merged back into `main`.

yinhwa commented:
Thank you @montoyamoraga and @limzykenneth for your quick responses! 😄 Just to share a bit more info : 

1. the changes in i18n-tracking.yml/ en.yml / es.yml / ko.yml / zh-Hans.yml were to add according translation lines for the newly added text info (on the newly submitted teaching resources) which I should've done manually (although I didn't know that i18n-tracking.yml could have been automatically done😅),

2. for main.css, I've only changed some parts in the teach section of it, but if it seemed to contain more changes, i think it's prolly due to the fact that the teach branch itself is outdated, as Kenneth mentioned?

3. @limzykenneth if by "rebase" you meant pulling the latest main branch in teach branch, and then add the same changes on teach page, I'll reopen the PR :)

Thank you both 🎈🎈🎇

limzykenneth commented:
@yinhwa If possible, I'd prefer to keep the `teach` branch up to date with the `main` branch so make future mergin easier, which means occasionally pulling and merging from `main` into `teach`.

yinhwa commented:
Sure @limzykenneth ! Will update this soon :) thanks again 🙏🙏

axwnet committed:
fixed typo: frameCount is a variable not a function

harneetsahi committed:
Style: replaced outlines with borders for improved accessibility

harneetsahi committed:
Style: fix search bar outline

raclim committed:
remove editor related images

raclim committed:
add updated editor images for intro tutorials

raclim committed:
update image paths and alt text for editor images in intro tutorials

ksen0 committed:
Update contributors list

ksen0 committed:
Merge pull request #804 from processing/update-contributors-main2

Update contributors list

davepagurek committed:
Revert "Update p5.js to v1.11.4"

This reverts commit 568a8b8ad38807423a8b6b933f73917aedbdd311.

davepagurek committed:
Rebuild contributor docs, contributors, reference, p5 version

davepagurek committed:
Update version

davepagurek committed:
Merge pull request #812 from processing/feat/perf

Fully unmount sketches that go out of frame

taylorvance committed:
Create p5.fps.yaml

taylorvance committed:
Add p5.fps image

taylorvance committed:
Larger fps img

davepagurek committed:
Add libraries, fix bug with old featured ones not getting cycled

davepagurek committed:
Merge pull request #819 from processing/more-libraries

Add libraries, fix bug with old featured ones not getting cycled

davepagurek committed:
Merge pull request #816 from taylorvance/main

FPS library

shivasankaran18 committed:
fix:removed linter warnings

Signed-off-by: shiva <shiva_05@outlook.com>

webermayank committed:
updated reference regex to not break on beta versions

webermayank committed:
updated regex to remove Unnecessary escape character

humanbydefinition committed:
Renamed `p5.ascify.yaml` to `p5.asciify.yaml`.

Updated description and `sourceURL`.

clairep94 committed:
fix .lowerCase error

clairep94 committed:
minimal error handling with no return value for errors

clairep94 committed:
add minimal error handling to other functions

clairep94 committed:
update community 10 sketches to be fetched with getCuratedSketches with no args and memoize getCurationSketches

clairep94 committed:
remove call to getSketchSize

clairep94 committed:
Add note

clairep94 committed:
refactor getSketch to first check for sketchId. with test console log

clairep94 committed:
destructure return from getSketch for legibility and revert title to come from getSketch

clairep94 committed:
test if getSketchSize is necessary

clairep94 committed:
remove comments about getSketcheSize, should be implemented after OP api is updated

clairep94 committed:
cleanup

ksen0 committed:
Merge pull request #838 from clairep94/patchfix_final

Resolves #831: Reduces calls to Open Processing API to prevent hitting rate limit errors

ksen0 committed:
Merge branch 'main' into fix/linkButton-searchBar-ouline

raclim committed:
update screenshots with consistent browser theme

raclim committed:
Merge branch 'main' into updated-editor-introduction

ksen0 committed:
Merge pull request #840 from raclim/updated-editor-introduction

Updated editor introduction

ksen0 committed:
Merge branch 'main' into main

ksen0 committed:
Merge pull request #834 from humanbydefinition/main

Updated `p5.asciify` add-on library content

ksen0 committed:
Merge branch 'main' into issue-771

ksen0 committed:
Merge branch 'main' into update-reference-regex-not-breat-Beta

ksen0 committed:
Merge branch 'main' into patch-1

limzykenneth committed:
Set iframe srcdoc attribute to blank to prevent iframe navigation

github-actions[bot] committed:
Update p5.js to v1.11.7

perminder-17 committed:
Merge branch 'main' into patch-1

perminder-17 committed:
Merge pull request #706 from notthewave/patch-1

fixed typo: frameCount is a variable not a function

limzykenneth committed:
Merge pull request #844 from limzykenneth/fix-reference-history

Set iframe srcdoc attribute to blank to prevent iframe navigation

ifenghm committed:
submission for shapeSVG library

ifenghm committed:
rename file

ksen0 committed:
Merge pull request #847 from irenelfeng/main

submission for shapeSVG library

dependabot[bot] committed:
Bump the npm_and_yarn group across 1 directory with 1 update

Bumps the npm_and_yarn group with 1 update in the / directory: [tar-fs](https://github.com/mafintosh/tar-fs).


Updates `tar-fs` from 2.1.1 to 3.0.9
- [Commits](https://github.com/mafintosh/tar-fs/compare/v2.1.1...v3.0.9)

Updates `tar-fs` from 3.0.5 to 3.0.9
- [Commits](https://github.com/mafintosh/tar-fs/compare/v2.1.1...v3.0.9)

---
updated-dependencies:
- dependency-name: tar-fs
  dependency-version: 3.0.9
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: tar-fs
  dependency-version: 3.0.9
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump prismjs in the npm_and_yarn group across 1 directory

Bumps the npm_and_yarn group with 1 update in the / directory: [prismjs](https://github.com/PrismJS/prism).


Updates `prismjs` from 1.29.0 to 1.30.0
- [Release notes](https://github.com/PrismJS/prism/releases)
- [Changelog](https://github.com/PrismJS/prism/blob/v2/CHANGELOG.md)
- [Commits](https://github.com/PrismJS/prism/compare/v1.29.0...v1.30.0)

---
updated-dependencies:
- dependency-name: prismjs
  dependency-version: 1.30.0
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump vite in the npm_and_yarn group across 1 directory

Bumps the npm_and_yarn group with 1 update in the / directory: [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite).


Updates `vite` from 5.1.6 to 5.4.19
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.4.19/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.19/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 5.4.19
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @babel/helpers in the npm_and_yarn group across 1 directory

Bumps the npm_and_yarn group with 1 update in the / directory: [@babel/helpers](https://github.com/babel/babel/tree/HEAD/packages/babel-helpers).


Updates `@babel/helpers` from 7.23.9 to 7.27.3
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.3/packages/babel-helpers)

---
updated-dependencies:
- dependency-name: "@babel/helpers"
  dependency-version: 7.27.3
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

aijun19 committed:
Update education-resources.mdx

The Animation Codebook (website) - A brief introduction to p5.js for those with an interest in animation principles.

aishwarya-ual committed:
Adding The Animation Codebook to Education Resources

ksen0 committed:
Merge pull request #858 from aijun19/addsite

Adding The Animation Codebook to Education Resources

ksen0 committed:
Merge branch 'main' into patch-1

ksen0 committed:
Merge pull request #857 from aijun19/patch-1

Update education-resources.mdx

ksen0 committed:
Merge pull request #820 from shivasankaran18/issue-771

Removed linter warnings

ksen0 committed:
Merge pull request #852 from processing/dependabot/npm_and_yarn/npm_and_yarn-2f20eee292

Bump the npm_and_yarn group across 1 directory with 1 update

ksen0 committed:
Create Accessibility Issue template

ksen0 committed:
Update 4.accessibility.yml

ksen0 committed:
Merge pull request #853 from processing/dependabot/npm_and_yarn/npm_and_yarn-46be7da7dc

Bump prismjs from 1.29.0 to 1.30.0 in the npm_and_yarn group across 1 directory

github-actions[bot] committed:
Update p5.js to v1.11.8

ksen0 committed:
Merge branch 'main' into update-reference-regex-not-breat-Beta

ksen0 committed:
Update reference.ts

ksen0 committed:
Merge pull request #855 from processing/dependabot/npm_and_yarn/npm_and_yarn-ec3f92bed1

Bump @babel/helpers from 7.23.9 to 7.27.3 in the npm_and_yarn group across 1 directory

ksen0 committed:
Merge branch 'main' into update-reference-regex-not-breat-Beta

ksen0 committed:
Merge pull request #824 from webermayank/update-reference-regex-not-breat-Beta

Updated reference regex to not break on beta versions

davepagurek committed:
Fix incorrect link for p5.woff2

davepagurek committed:
Merge pull request #861 from processing/woff2-fix

Fix incorrect link for p5.woff2

ksen0 committed:
Merge pull request #854 from processing/dependabot/npm_and_yarn/npm_and_yarn-8691e273c3

Bump vite from 5.1.6 to 5.4.19 in the npm_and_yarn group across 1 directory

ksen0 committed:
Merge pull request #860 from ksen0/main

Add an Issue template for reporting accessibility issues

ksen0 committed:
Update placeholders in 4.accessibility.yml

ksen0 committed:
Merge pull request #864 from processing/ksen0-patch-2

Update placeholders in 4.accessibility.yml

shivasankaran18 committed:
feat:added welcomebot

ksen0 committed:
Update 4.accessibility.yml to remove extra "steps to reproduce field"

Fixes an accidental inclusion of an unnecessary field

ksen0 committed:
Merge pull request #867 from processing/ksen0-patch-2

Update 4.accessibility.yml to remove extra "steps to reproduce field"

AshmitSherigar committed:
Add missing H1 to / route for proper heading hierarchy and accessibility

AshmitSherigar committed:
Fix heading levels on /download page (H3 → H2 for accessibility)

ksen0 committed:
Merge pull request #821 from shivasankaran18/issue-817

Add Welcome Bot configuration for first-time contributors

Yo7ee committed:
Fixes #882

tsengwoody committed:
resolve #872 on tutorial page only; other pages pending fixes

MissTipo committed:
chore(content): replace old-site message with open call submission promo

Signed-off-by: MissTipo <dorine.a.tipo@gmail.com>

davepagurek committed:
Merge pull request #893 from MissTipo/chore/open-call-promo

replace old-site alert with open call submission promo

eslteacher902010 committed:
Add aria-labels and accessibility improvements to buttons and navigation components

perminder-17 committed:
Merge branch 'main' into fix-heading-structure

perminder-17 committed:
Merge pull request #883 from AshmitSherigar/fix-heading-structure

Fix heading structure for improved accessibility

perminder-17 committed:
Merge branch 'main' into issue#872

perminder-17 committed:
Merge pull request #890 from coseeian/issue#872

resolve #872 on tutorial page only; other pages pending fixes

iamniky committed:
Update abs.mdx

Clarify abs() reference explanation

perminder-17 committed:
Merge pull request #899 from iamniky/edit-abs-description

Update abs.mdx

reshma045 committed:
Fixes #887

tsengwoody committed:
Add screen reader announcement message after CopyCodeButton completes action

tsengwoody committed:
Refactor screen reader announcement message into a reusable hook

ksen0 committed:
Merge pull request #891 from coseeian/issue#882

Fixes Screen reader and visual presentation are inconsistent. (issue#882)

tsengwoody committed:
Add screen reader announcements for Reset, Run, and Stop sketch actions within code blocks

ksen0 committed:
Merge branch 'main' into fix/linkButton-searchBar-ouline

eslteacher902010 committed:
Fix JSX indentation in JumpToLinks

eslteacher902010 committed:
Fix JSX indentation in MainNavLinks toggle

tsengwoody committed:
fix(useLiveRegion): clear pending timers to prevent race conditions & leaks

ksen0 committed:
Merge pull request #905 from coseeian/issue#866

Add aria-live status messages for button actions to improve screen reader feedback (fixes issue #866)

github-actions[bot] committed:
Update p5.js to v1.11.9

ksen0 committed:
Merge pull request #910 from eslteacher902010/aria-labels-pr-2

Aria labels pr 2

reshma045 committed:
fixed unnecessary inversion of photos

lirenjie95 committed:
Fix: Use import for fes.svg instead of copying to public

ksen0 committed:
Merge branch 'main' into fix/linkButton-searchBar-ouline

ksen0 committed:
Merge pull request #766 from harneetsahi/fix/linkButton-searchBar-ouline

Fixed LinkButton and SearchForm outline

reshma045 committed:
Merge pull request #1 from lirenjie95/fix-fes-diagram-import

Fix: Use import for fes.svg instead of copying to public

ksen0 committed:
Merge branch 'main' into issue887-branch

ksen0 committed:
Merge pull request #902 from reshma045/issue887-branch

Fix: Improve image visibility in dark mode by inverting images

Kakashi786 committed:
Add Hindi, Spanish, Korean translations for tutorial section headings

MissTipo committed:
chore(content): replace open call submission promo with the old-site banner alert

Signed-off-by: MissTipo <dorine.a.tipo@gmail.com>

davepagurek committed:
Merge pull request #932 from MissTipo/chore/open-call-promo

replace the open call submission promo with the old-site banner alert

ksen0 committed:
Merge branch 'main' into fix-i18n-translations

Kakashi786 committed:
Fix words based on review comments

Kakashi786 committed:
Merge branch 'fix-i18n-translations' of https://github.com/Homaid/p5.js-website into fix-i18n-translations

MissTipo committed:
feat(community): expand curated sketches to include 2025 set

- Updated `getCurationSketches` in `src/api/OpenProcessing.ts` to:
  - Remove hardcoded fetch limit to retrieve all curated sketches
  - Support fetching from both 2024 and 2025 curation IDs
  - Add `curation` property to each sketch for filtering and ordering
- Updated `src/pages/community.astro` to:
  - Use a priority ID list to select and order featured sketches
  - Ensure no duplicate sketches appear in the display
- Added high-resolution images for 2025 curated sketches

Signed-off-by: MissTipo <dorine.a.tipo@gmail.com>

MissTipo committed:
Remove redundant duplicate filter

Signed-off-by: MissTipo <dorine.a.tipo@gmail.com>

perminder-17 committed:
Update src/content/ui/ko.yaml

perminder-17 committed:
Merge pull request #931 from Homaid/fix-i18n-translations

Add missing translations for tutorial section headings

perminder-17 committed:
fixing-build-failure on deployment

ksen0 committed:
Merge pull request #940 from perminder-17/main

Add missing tutorialCategories.education-resources to en.yaml

davepagurek committed:
Merge pull request #942 from MissTipo/community-sketches-main

Sync curated sketches update and filter cleanup from 2.0 into main

adi-ray committed:
Fix Markdown backtick escaping in Variables and Change tutorial

perminder-17 committed:
Merge pull request #944 from adi-ray/fix-markdown

Fix Markdown backtick escaping in Variables and Change tutorial

adi-ray committed:
docs: Update reference generation instructions to p5.js 2.0 fork workflow

perminder-17 reviewed (changes requested):
Hi @adi-ray , the reference generation instructions was supposed to be updated in the 2.0 branch.

Actually we have deployed the 2.x version with `2.0` branch of `p5.js-website` and all 1.x versions are deployed with the `main` branch. Can you please re-target your PR according to the `2.0` branch?

Steps:
1. In the PR, click Edit -> change the base from main to 2.0 -> Save.

Locally:
```
git fetch origin
git checkout update-doc
git rebase origin/2.0
git push --force-with-lease
```

adi-ray commented:
> Hi @adi-ray , the reference generation instructions was supposed to be updated in the 2.0 branch.
> 
> Actually we have deployed the 2.x version with `2.0` branch of `p5.js-website` and all 1.x versions are deployed with the `main` branch. Can you please re-target your PR according to the `2.0` branch?
> 
> Steps:
> 
> 1. In the PR, click Edit -> change the base from main to 2.0 -> Save.
> 
> Locally:
> 
> ```
> git fetch origin
> git checkout update-doc
> git rebase origin/2.0
> git push --force-with-lease
> ```

Hi! Thanks for the clarification about targeting the 2.0 branch. I'm running into some rebase conflicts.

Lemme just create a fresh PR directly against the 2.0 branch instead.
